67_RayQueryGeometry/main.cpp

```diff
@@ -158,56 +106,33 @@ class RayQueryGeometryApp final : public examples::SimpleWindowedApplication, pu
 			m_surface->recreateSwapchain();
 
 			// create output images
-			for (uint32_t i = 0; i < MaxFramesInFlight; i++)
-			{
-				IGPUImage::SCreationParams imgInfo;
-				imgInfo.format = asset::EF_R16G16B16A16_SFLOAT;
-				imgInfo.type = IGPUImage::ET_2D;
-				imgInfo.extent.width = WIN_W;
-				imgInfo.extent.height = WIN_H;
-				imgInfo.extent.depth = 1u;
-				imgInfo.mipLevels = 1u;
-				imgInfo.arrayLayers = 1u;
-				imgInfo.samples = asset::ICPUImage::ESCF_1_BIT;
-				imgInfo.flags = static_cast<asset::IImage::E_CREATE_FLAGS>(0u);
-				imgInfo.usage = core::bitflag(asset::IImage::EUF_STORAGE_BIT) | asset::IImage::EUF_TRANSFER_SRC_BIT;
-
-				auto image = m_device->createImage(std::move(imgInfo));
-				auto imageReqs = image->getMemoryReqs();
-				imageReqs.memoryTypeBits &= m_device->getPhysicalDevice()->getDeviceLocalMemoryTypeBits();
-				auto imageMem = m_device->allocate(imageReqs, image.get());
-
-				IGPUImageView::SCreationParams imgViewInfo;
-				imgViewInfo.image = std::move(image);
-				imgViewInfo.format = asset::EF_R16G16B16A16_SFLOAT;
-				imgViewInfo.viewType = IGPUImageView::ET_2D;
-				imgViewInfo.flags = static_cast<IGPUImageView::E_CREATE_FLAGS>(0u);
-				imgViewInfo.subresourceRange.aspectMask = IImage::E_ASPECT_FLAGS::EAF_COLOR_BIT;
-				imgViewInfo.subresourceRange.baseArrayLayer = 0u;
-				imgViewInfo.subresourceRange.baseMipLevel = 0u;
-				imgViewInfo.subresourceRange.layerCount = 1u;
-				imgViewInfo.subresourceRange.levelCount = 1u;
-
-				outHDRImageViews[i] = m_device->createImageView(std::move(imgViewInfo));
-			}
-
-			{
-				IGPUBuffer::SCreationParams params;
-				params.usage = IGPUBuffer::EUF_UNIFORM_BUFFER_BIT | IGPUBuffer::EUF_TRANSFER_DST_BIT | IGPUBuffer::EUF_INLINE_UPDATE_VIA_CMDBUF;
-				params.size = sizeof(SCameraParameters);
-				cameraBuffer = createBuffer(params);
-			}
+			outHDRImage = m_device->createImage({
+				{
+					.type = IGPUImage::ET_2D,
+					.samples = asset::ICPUImage::ESCF_1_BIT,
+					.format = asset::EF_R16G16B16A16_SFLOAT,
+					.extent = {WIN_W, WIN_H, 1},
+					.mipLevels = 1,
+					.arrayLayers = 1,
+					.flags = IImage::ECF_NONE,
+					.usage = core::bitflag(asset::IImage::EUF_STORAGE_BIT) | asset::IImage::EUF_TRANSFER_SRC_BIT
+				}
+			});
+			if (!outHDRImage || !m_device->allocate(outHDRImage->getMemoryReqs(), outHDRImage.get()).isValid())
+				return logFail("Could not create HDR Image");
```

Comment:
you don't need a HDR image for this example, you can probably store straight to swapchain image if you want